### PR TITLE
Code restructuring

### PR DESCRIPTION
**Pull request by AlexeyBarkow**

### CODE REVIEW

File: app/API/getCity.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const username = `aliaksei_barkou`;
```
Comment (hamanovich) [inline app/API/getCity.js]: for static const use UPPER_CASE

File: app/API/weatherAPI.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// import fetchSimplified from './fetch.simplified.js';
```
Comment (hamanovich) [inline app/API/weatherAPI.js]: remove commented lines

File: app/API/weatherAPI.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import axios from 'axios';
+
+const DEFAULT_API_KEY = 'e370193d5122935a572140b76ba0a173';
+
+function getWeatherTemplate(city, apiKey = DEFAULT_API_KEY) {
+    return `api.openweathermap.org/data/2.5/weather?q=${city}&key=${DEFAULT_API_KEY}`;
```
Comment (hamanovich) [inline app/API/weatherAPI.js]: api.openweathermap.org - is also good way move to constant

File: app/actions/actions.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import * as types from './types';
+import getWeatherAjax from '../API/weatherAPI';
+
+
+export function changeCity(city) {
+    return {
+        type: types.SET_CITY,
+        city,
```
Comment (hamanovich) [inline app/actions/actions.js]: one of the good solution use  default syntax for actions: return{ type: type, payload: ....}

File: app/components/Button.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+function Button({
+    children,
+    onClickHandler,
+    className,
+    href,
+    disabled,
+}) {
+    const classesToPass = `btn btn-default ${className}`;
+    return (
```
Comment (hamanovich) [inline app/components/Button.js]: add empty line before return statement

File: app/components/Button.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        href !== ''
+        ?
+        <Link
+          disabled={disabled}
+          className={classesToPass}
+          to={href}
+          onClick={onClickHandler}
+        >
```
Comment (hamanovich) [inline app/components/Button.js]: close > in the previous line

File: app/components/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import React, { PropTypes } from 'react';
+import Button from './Button';
+// import MenuItem from './MenuItem';
+import ButtonGroup from './ButtonGroup';
+// import ButtonDropDown from '../containers/ButtonDropDown';
+// import weatherAPI from '../API/weatherAPI';
```
Comment (hamanovich) [inline app/components/Navbar.js]: remove unused imports

File: app/reducers/index.js [hunk 8]
```diff
@@ -1,19 +1,33 @@
...
+};
+
+function weatherApp(state = initialState, action) {
     switch (action.type) {
-        case types.FILTER:
-            return action.filter
+        case types.SET_CITY:
+            return {
+                city: action.city,
+                ...state,
```
Comment (hamanovich) [inline app/reducers/index.js]: no needed to use comma after last line

File: app/reducers/index.js [hunk 9]
```diff
@@ -1,19 +1,33 @@
...
+            };
+        case types.UPDATE_WEATHER_INFO:
+            return {
+                weather: action.weather,
+                ...state,
```
Comment (hamanovich) [inline app/reducers/index.js]: also comma is redundunt

File: dist/index.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/latest/css/bootstrap.min.css">
+</head>
+<body>
+    <div id="root"></div>
+<script type="text/javascript" src="./polyfills.min.js"></script><script type="text/javascript" src="./vendor.min.js"></script><script type="text/javascript" src="./main.min.js"></script></body>
```
Comment (hamanovich) [inline dist/index.html]: instead of adding all js files, include webpack chunk plugin. As I've showed you earlier.

File: webpack.config.js [hunk 11]
```diff
@@ -31,20 +29,31 @@ module.exports = {
...
+        {
+            test: /\.js$/,
+            exclude: /node_modules/,
+            loader: 'babel'
+        }, {
+            test: /\.json?$/,
+            loader: 'json'
+        }, {
+            test: /\.scss$/,
+            loader: 'style!css!sass?modules&localIdentName=[name]---[local]---[hash:base64:5]'
```
Comment (hamanovich) [inline webpack.config.js]: i suggest you re-write one-line syntax for loader to array with included loaders

File: dist/index.html (hunk 10 above)
Comment (AlexeyBarkow) [inline dist/index.html]: these file is generated programmatically, I've just forgotten to remove the dist folder from git

File: app/reducers/index.js (hunk 9 above)
Comment (AlexeyBarkow) [inline app/reducers/index.js]: it's a point in suggested requirements, so default eslint settings demand this comma

File: app/components/Button.js (hunk 6 above)
Comment (AlexeyBarkow) [inline app/components/Button.js]: This one is in react styleguides

**Pull request merged**